infrastructure/src/main/java/org/corfudb/infrastructure/management/PollReport.java

```diff
@@ -21,43 +23,78 @@
  */
 @Data
 @Builder
+@Slf4j
 public class PollReport {
 
     @Default
-    private long pollEpoch = Layout.INVALID_EPOCH;
+    private final long pollEpoch = Layout.INVALID_EPOCH;
+
     /**
-     * Contains list of servers successfully connected with current node.
-     * It doesn't contain nodes answered with WrongEpochException.
+     * A map of nodes answered with {@link WrongEpochException}.
      */
     @Default
-    private final ImmutableSet<String> connectedNodes = ImmutableSet.of();
+    private final ImmutableMap<String, Long> wrongEpochs = ImmutableMap.of();
+
     /**
-     * Contains list of servers disconnected from this node. If the node A can't ping node B then node B will be added
-     * to failedNodes list.
+     * Current cluster state, collected by a failure detector
      */
-    @Default
-    private final ImmutableSet<String> failedNodes = ImmutableSet.of();
+    @NonNull
+    private final ClusterState clusterState;
+
     /**
-     * A map of nodes answered with {@link WrongEpochException}.
+     * List of responsive servers in a current layout
      */
-    @Default
-    private final ImmutableMap<String, Long> wrongEpochs = ImmutableMap.of();
+    @NonNull
+    private final List<String> activeLayoutServers;
```

Comment:
Rename it to activeServers or activeNodes.
This can be confused with the layout server component in Corfu.

Reply:
renamed to `responsiveServers`